lib/autoupdate/agent/updater.go

```diff
@@ -173,12 +182,12 @@ type Updater struct {
 	Installer Installer
 	// Process manages a running instance of Teleport.
 	Process Process
-	// Setup installs the Teleport updater service using the linked installation.
-	Setup func(ctx context.Context) error
-	// Revert installs the Teleport updater service using the running installation.
-	Revert func(ctx context.Context) error
-	// Teardown removes all traces of the updater and all managed installations.
-	Teardown func(ctx context.Context) error
+	// ExecSetup execs teleport-update to verify and reload the installation.
+	ExecSetup func(ctx context.Context, restart bool) error
+	// SetupConfig installs the Teleport updater service using the running installation.
+	SetupConfig func(ctx context.Context) error
+	// TeardownConfig removes all traces of the updater and all managed installations.
+	TeardownConfig func(ctx context.Context) error
```

Comment:
Those names are not very explicit about what those functions are doing, especially the `*Config` part.

Reply:
```suggestion
	// VerifyAndReload execs teleport-update to verify and reload the installation.
	VerifyAndReload func(ctx context.Context, restart bool) error
	// SetupUpdater installs the Teleport updater service using the running installation.
	SetupUpdater func(ctx context.Context) error
	// Teardown removes all traces of the updater and all managed installations.
	Teardown func(ctx context.Context) error
```

Reply:
👍 Made these names more descriptive and added better godoc.

(Note that `ExecSetup` re-execs the binary with the `setup` subcommand, while the others just execute `Namespace` methods.)